File: src/com/nimbusly/isometrix/Point.java
package com.nimbusly.isometrix;

public class Point {
	public int x, y;
	
	public Point(int x, int y) {
		set(x, y);
	}

	public Point(Point p) {
		set(p.x, p.y);
	}
	
	public Point set(int x, int y) {
		this.x = x;
		this.y = y;
		return this;
	}
	
	public Point offset(int dx, int dy) {
		return set(x + dx, y + dy);
	}
	
	@Override
	public String toString() {
		return "Point(" + x + "," + y + ")";
	}
	
	@Override
	public boolean equals(Object other) {
		if (this == other) return true;
		if (!(other instanceof Point)) return false;
		Point otherPoint = (Point) other;
		return this.x == otherPoint.x && this.y == otherPoint.y;
	}
	
	@Override
	public int hashCode() {
		return x ^ y;
	}

	public double distanceTo(Point other) {
		int dx = x - other.x;
		int dy = y - other.y;
		return Math.sqrt(dx*dx + dy*dy);
	}
	
	
}

File: src/com/nimbusly/isometrix/Point.scala
package com.nimbusly.isometrix

class Point(var x : Int = 0, var y : Int = 0) {

	def this(p : Point) = this(p.x, p.y)
	
	def set(x : Int, y : Int) : Point = {
		this.x = x
		this.y = y
		return this
	}
	
	def offset(dx : Int, dy : Int) : Point = {
		return set(x + dx, y + dy)
	}
	
	override def toString : String = "Point(" + x + "," + y + ")"
	
	override def equals(other : Any) : Boolean = {
		other match {
			case p:Point => this.x == p.x && this.y == p.y
			case _ => false
		}
	}
	
	override def hashCode : Int = x ^ y

	def distanceTo(other : Point) : Int = {
		val dx: Int = x - other.x
		val dy: Int = y - other.y
		math.round(math.sqrt(dx*dx + dy*dy).asInstanceOf[Float])
	}
	
}